src/Umbraco.Infrastructure/Runtime/FileSystemMainDomLock.cs

```diff
@@ -0,0 +1,147 @@
+
+            return Task.FromResult(false);
+        }
+
+        // Create a long running task to poll to check if anyone has created a lock release file.
+        public Task ListenAsync()
+        {
+            if (_listenForReleaseSignalFileTask != null)
+            {
+                return _listenForReleaseSignalFileTask;
+            }
+
+            _listenForReleaseSignalFileTask = Task.Factory.StartNew(
+                ListeningLoop,
+                _cancellationTokenSource.Token,
+                TaskCreationOptions.LongRunning,
+                TaskScheduler.Default);
+
+            return _listenForReleaseSignalFileTask;
+        }
+
+        public void Dispose()
+        {
+            _lockFileStream?.Close();
+            _lockFileStream = null;
+        }
+
+        private void CreateLockReleaseFile()
+        {
+            try
+            {
+                // Dispose immediately to release the file handle so it's easier to cleanup in any process.
+                using FileStream releaseFileStream = File.Open(_releaseSignalFilePath, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.ReadWrite);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Unexpected exception attempting to create lock release signal file {file}", _releaseSignalFilePath);
+            }
+        }
+
+        private void DeleteLockReleaseFile()
+        {
+            List<Exception> encounteredExceptions = new();
+            for (var i = 0; i < s_maxTriesRemovingLockReleaseSignalFile; i++)
+            {
+                try
+                {
+                    File.Delete(_releaseSignalFilePath);
+                    return;
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Unexpected exception attempting to delete release signal file {file}", _releaseSignalFilePath);
+                    encounteredExceptions.Add(ex);
+                    Thread.Sleep(500 * (i + 1));
+                }
+            }
+
+            throw new ApplicationException($"Failed to remove lock release signal file {_releaseSignalFilePath}", new AggregateException(encounteredExceptions));
+        }
+
+        private void ListeningLoop()
+        {
+            while (true)
+            {
+                if (_cancellationTokenSource.IsCancellationRequested)
+                {
+                    _logger.LogDebug("ListenAsync Task canceled, exiting loop");
+                    return;
+                }
+
+                if (File.Exists(_releaseSignalFilePath))
+                {
+                    _logger.LogDebug("Found lock release signal file, releasing lock on {lockFilePath}", _lockFilePath);
+                    _lockFileStream?.Close();
+                    _lockFileStream = null;
+                    break;
+                }
+
+                Thread.Sleep(2000);
```

Comment:
Should this interval be configurable? The default polling interval used by the `PhysicalFileProvider` seems to be [4 seconds](https://github.com/dotnet/runtime/blob/3ae87395f638a533f37b8e3385f6d3f199a72f4f/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs#L31). Remember that this causes an I/O lookup every n seconds, which might require going over the network for remote filesystems (like used on Azure)...

Reply:
This doesn't make your point any less valid, but have you seen what we do every two seconds with SqlMainDomLock? https://github.com/umbraco/Umbraco-CMS/blob/de4b3af28f091c4dec61df5eb17203eb98860528/src/Umbraco.Infrastructure/Runtime/SqlMainDomLock.cs#L233

Reply:
On azure if this is hitting UNC share it means config is invalid as docs state you should be using EnvironmentTemp for LocalTempStorageLocation

Reply:
Resolved #12053